assets/scripts/segments/segment-dict.js

```diff
@@ -0,0 +1,233 @@
+  let graphicsInfo
+
+  if (Array.isArray(target) && Array.isArray(source)) {
+    graphicsInfo = [...target, ...source]
+  } else if (Array.isArray(target)) {
+    graphicsInfo = [...target, source]
+  } else if (Array.isArray(source)) {
+    graphicsInfo = [ target, ...source ]
+  } else {
+    graphicsInfo = (target === source) ? target : [ target, source ]
+  }
+
+  return graphicsInfo
+}
+
+/**
+ * Merges the array of graphic definition objects needed to draw the segment into a single object.
+ *
+ * @param {Array} variantGraphics
+ * @returns {object} graphics
+ */
+function mergeVariantGraphics (variantGraphics) {
+  return variantGraphics.reduce((graphics, variantInfo) => {
+    Object.keys(variantInfo).forEach((key) => {
+      const target = graphics[key]
+      const source = variantInfo[key]
+
+      if (target) {
+        graphics[key] = appendVariantSprites(target, source)
+      } else {
+        graphics[key] = source
+      }
+    })
+
+    return graphics
+  }, {})
+}
+
+/**
+ * Collects all rules associated with any items that make up the component group into a single object.
+ *
+ * @param {Array} groupItems - items that make up the component group in shape of [{ id, variants }]
+ * @param {object} componentGroupInfo - definition of each item from `SEGMENT_INFO` in shape of { id: { characteristics, rules, variants } }
+ * @returns {object} componentGroupRules
+ */
+function getComponentGroupRules (groupItems, componentGroupInfo) {
+  const componentGroupRules = groupItems.reduce((obj, item) => {
+    const { id, variants } = item
+
+    let groupItemRules = {}
+
+    if (componentGroupInfo[id] && componentGroupInfo[id].rules) {
+      groupItemRules = { ...componentGroupInfo[id].rules }
+    }
+
+    const groupItemVariants = componentGroupInfo[id] && componentGroupInfo[id].variants
+
+    Object.keys(variants).forEach((variantName) => {
+      if (groupItemVariants[variantName] && groupItemVariants[variantName].rules) {
+        groupItemRules = { ...groupItemRules, ...groupItemVariants[variantName].rules }
+      }
+    })
+
+    return { ...obj, ...groupItemRules }
+  }, {})
+
+  return Object.keys(componentGroupRules).length && componentGroupRules
+}
+
+/**
+ * Maps the old segment data model to the new segment data model and returns the graphic sprites necessary
+ * to draw the segment as well as any rules to follow, e.g. `minWidth` based on the `type` and `variant`.
+ *
+ * @param {string} type
+ * @param {string} variant
+ * @returns {object} variantInfo - returns an object in the shape of { graphics, ...rules }
+ */
+function getSegmentVariantInfo (type, variant) {
+  const segmentLookup = getSegmentLookup(type, variant)
+
+  if (!segmentLookup) {
+    return SEGMENT_UNKNOWN_VARIANT
+  }
+
+  let variantInfo = { graphics: [] }
+
+  // 1) Loop through each component group that makes up the segment.
+  variantInfo = Object.entries(segmentLookup).reduce((variantInfo, componentGroup) => {
```

Comment:
Instead of defining `variantInfo` with `let`, then passing it into `.reduce()` and assigning to itself -- would this work:

```js
const variantInfo = Object.entries(segmentLookup).reduce((variantAccumulator, componentGroup) => {
   // ...
}, { graphics: [] })
```